### ISSUE TITLE
Implement merchant buy/sell menus.

### ISSUE THREAD

**Issue by ZaneDubya** [ZaneDubya/UltimaXNA#138]
As a player, I want to be able to interact with the merchants by buying and selling from them. The interface for this feature should resemble the legacy client's.

**Comment by robertdeclaux** [comment]
:+1:

**Comment by ZaneDubya** [comment]
@robertdeclaux This is definitely on the todo list. Reimplementing the legacy gumps is really, really tedious for me. But I will absolutely get this done before starting on the next milestone. :)

**Comment by ZaneDubya** [comment]
Okay, last issue in Milestone 0.6. Let's get this done!

**Comment by robertdeclaux** [comment]
Great! It will be nearly playable.

**Comment by ZaneDubya** [comment]
So I'm not going to do a complete recreation of these menus. They're really terrible. But my implementation will have the same functionality.

**Comment by ZaneDubya** [comment]
I have vendor shop data loading correctly now. Off to a good start!
![screen shot 2015-08-03 at 9 11 03 am](https://cloud.githubusercontent.com/assets/7041719/9039255/d198088a-39bf-11e5-88d9-d8c5e357a0dc.png)

**Comment by ZaneDubya** [comment]
A little closer...
![ss1](https://cloud.githubusercontent.com/assets/7041719/9043568/0869dafc-39dc-11e5-9483-0f1cedfcb5c1.jpg)

**Comment by ZaneDubya** [comment]
Finished! Would someone please sanity check me and make sure there are no obvious errors.

**Comment by robertdeclaux** [comment]
I will test it as soon as i got home

**Comment by ZaneDubya** [comment]
Note that you can't yet use 'vendor buy' - see #249. But the context menu works.

**Comment by robertdeclaux** [comment]
It is working with little problems (on sphereserver)
When i open buy menu it opens blank.
![ekran goruntusu 1](https://cloud.githubusercontent.com/assets/11157985/9085591/cb456f7c-3b86-11e5-8b02-389a61c168d0.png)

Then it gets data by click right on it
![ekran goruntusu 3](https://cloud.githubusercontent.com/assets/11157985/9085605/e3fedb34-3b86-11e5-9bc1-538048c19ed7.png)

Then when i confirm, It shows "ERROR UNKOWN COMMAND 3", but i can buy items.
![ekran goruntusu 4](https://cloud.githubusercontent.com/assets/11157985/9085627/171e85f0-3b87-11e5-9f3c-3820e4851c3b.png)

**Comment by ZaneDubya** [comment]
Very interesting. Maybe I can trouble shoot that this weekend as well..

**Comment by ZaneDubya** [comment]
@robertdeclaux "Then it gets data by click right on it" Do you mean you right click on the menu and then it shows the items you can buy? Would you try moving the empty gump - is there a gump full of items to buy behind it?

**Comment by robertdeclaux** [comment]
Sorry i inspected it wrong, when i try to open listing, it opens 2 menus one blank on the top and one filled on the back, when i right click on it i just close the blank one.

**Comment by ZaneDubya** [comment]
Looks like I need to fix up the text spacing as well... This will need some new html spacing attributes (I'm thinking float?)

EDIT: moved to #257.

**Comment by ZaneDubya** [comment]
Robert's problem involves sphere sending an empty 'vendor buy list' (possibly for the second closest vendor? Not sure).

Solved by adding this sanity checking to VendorBuyGump:

```
// sanity checking
            if (!(vendorBackpack is Container) || ((vendorBackpack as Container).Contents.Count <= 0) || (packet.Items.Count <= 0))
            {
                Dispose();
                return;
            }
```

Also will need to check if a control is disposed in UIService.AddControl.

Also should add some sanity checking to make sure AControl.Update isn't called when the control has been disposed in its initialize routine.

**Comment by ZaneDubya** [comment]
With that fix, Milestone 0.6 is complete!
